### PR TITLE
K8s _source etc. Sumo meta additions

### PR DESCRIPTION
**Pull request by pmm-sumo**
**Description:** 

Gets closer to having 1:1 match `sumologic-kubernetes-collection` plugins in metadata tagging and rules for K8S. This also enables more flexibility in how the tag names are being configured, enabling custom taxonomy

### CODE REVIEW

File: processor/k8sprocessor/kube/owner.go [hunk 1]
```diff
@@ -127,6 +131,40 @@ func newOwnerProvider(
...
+	op.cacheMutex.Lock()
+	op.namespaces[namespace.Name] = &namespace
+	op.cacheMutex.Unlock()
+}
+
+func (op *OwnerCache) deleteNamespace(obj interface{}) {
+	namespace := obj.(api_v1.Namespace)
+	op.cacheMutex.Lock()
+	delete(op.namespaces, namespace.Name)
+	op.cacheMutex.Unlock()
```
Comment (fdiaslew) [inline processor/k8sprocessor/kube/owner.go:145]: There might be very little risk here and maybe not worth the trouble, but I see a lot of the following pattern when working with locks:

```
op.cacheMutex.Lock()
defer op.cacheMutex.Unlock()
....
```

Comment (piotr-sumo) [reply processor/k8sprocessor/kube/owner.go:145]: 👍

File: processor/sourceprocessor/source_processor.go [hunk 2]
```diff
@@ -41,27 +64,132 @@ type attributeFiller struct {
...
+	if regex == "" {
+		return nil
+	}
+
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		log.Fatalf("Cannot compile reguler expression: %s Error: %v\n", regex, err)
+	}
+
+	return re
```
Comment (piotr-sumo) [inline processor/sourceprocessor/source_processor.go:102]: What about returning nil regex when error happens? Will `regexp.Compile(regex)` return nil in the first element of the tuple?

File: processor/sourceprocessor/source_processor.go [hunk 3]
```diff
@@ -41,27 +64,132 @@ type attributeFiller struct {
...
 
+func (stp *sourceTraceProcessor) fillOtherMeta(atts pdata.AttributeMap) {
+	atts.UpsertString("_source", stp.source)
+	if stp.collector != "" {
+		atts.UpsertString("_collector", stp.collector)
+	}
+}
+
+func (stp *sourceTraceProcessor) isFilteredOut(atts pdata.AttributeMap) bool {
+	// TODO: This is quite inefficient when done for each package (ore even more so, span) separately.
```
Comment (piotr-sumo) [inline processor/sourceprocessor/source_processor.go:154]: Do we have a Jira ticket for this TODO?

File: processor/sourceprocessor/source_processor_test.go [hunk 4]
```diff
@@ -18,23 +18,21 @@ import (
 	"context"
 	"testing"
 
-	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
 	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
 	"github.com/stretchr/testify/assert"
 )
 
+func createConfig() *Config {
```
Comment (piotr-sumo) [inline processor/sourceprocessor/source_processor_test.go:25]: nice tests 👍

File: processor/sourceprocessor/source_processor.go (hunk 2 above)
Comment (pmm-sumo) [reply processor/sourceprocessor/source_processor.go:104]: I believe that in such case `log.Fatalf` will cause the application to exit, let me check that

Comment (pmm-sumo) [reply processor/sourceprocessor/source_processor.go:104]: Yeap:
```
2020/05/15 14:04:25 Cannot compile reguler expression: [ Error: error parsing regexp: missing closing ]: `[`

Process finished with exit code 1
```

Comment (pmm-sumo) [reply processor/sourceprocessor/source_processor.go:104]: (Just a small typo to fix in message) :)

File: processor/sourceprocessor/source_processor.go (hunk 3 above)
Comment (pmm-sumo) [reply processor/sourceprocessor/source_processor.go:156]: TRACING-542

**Pull request merged**